### PR TITLE
Add a top level API

### PR DESCRIPTION
**Pull request by kevindew**
This PR aims to make it easier to use Jsonnet in Ruby projects by adding a simpler API.

It adds two module functions to `Jsonnet` of `.parse` and `.load` (naming inspired by [Ruby JSON](https://ruby-doc.org/stdlib-2.4.0/libdoc/json/rdoc/JSON.html)) which initialise the `Jsonnet::VM` and run the output through `JSON.parse`. This gives you a simpler, nicer interface to load in Jsonnet files.

It also adds usage instructions for the README

### CODE REVIEW

File: lib/jsonnet/evaluator.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+module Jsonnet
+  class Evaluator
```
Comment (yugui) [inline lib/jsonnet/evaluator.rb]: * What is this class is for?
   I don't have a clear understanding of use cases of instances of this class. So I suspect if this class is just redundant.
* Also `Evaluator.from_snippet` sounds as if it returns an instance of `Evaluator`.  I guess this confusion is also coming from the same issue of redundancy.
* Feel free to just add this kind of higher-level convenient methods to `VM` class if necessary.

File: lib/jsonnet/evaluator.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    def self.from_snippet(snippet, options = {})
+      snippet_check = ->(key, value) { key.to_s.match(/^filename|multi$/) }
+      snippet_options = options.select &snippet_check
+      vm_options = options.reject &snippet_check
+      self.new(vm_options).snippet(snippet, snippet_options)
+    end
+
+    def self.from_file(filename, options = {})
```
Comment (yugui) [inline lib/jsonnet/evaluator.rb]: ditto.

File: lib/jsonnet.rb [hunk 3]
```diff
@@ -1,7 +1,49 @@
...
+  #                                  import_callback, gc_growth_triger,
+  #                                  gc_min_objects, max_stack, max_trace
+  # @param [Hash]    json_options Options supported by {JSON.parse}[http://www.rubydoc.info/github/flori/json/JSON#parse-class_method]
+  # @return [Hash] The JSON representation as a hash
+  # @raise [UnsupportedOptionError] Raised when an option passed is unsupported by Jsonnet::VM
+  #
+  # @note This method runs Jsonnet::VM#evaluate and runs the string
+  #       output through {JSON.parse}[http://www.rubydoc.info/github/flori/json/JSON#parse-class_method]
+  #       so those should be looked at for furhter details
+  def parse(jsonnet, jsonnet_options: {}, json_options: {})
```
Comment (kevindew) [inline lib/jsonnet.rb]: Sure evaluate sounds like a good idea, happy to change it to that.

File: lib/jsonnet/evaluator.rb (hunk 1 above)
Comment (kevindew) [inline lib/jsonnet/evaluator.rb]: The purpose of this class is mostly to take options in a form of hash (which is what the top level API can be provided with) and know how to apply them for the VM, whether they be instance methods or method arguments.

It's meant as a private class that the top level API uses.

I can look into adding these to the VM, see if they don't add too much bulk.

Comment (yugui) [inline lib/jsonnet/evaluator.rb]: How about something like this?
https://github.com/yugui/ruby-jsonnet/commit/4eca92cd7b87c2f1a1d6749266d8015f5d730448

Comment (kevindew) [inline lib/jsonnet/evaluator.rb]: Yup that looks good to me 👍 

I'm a bit confused about the need for: https://github.com/yugui/ruby-jsonnet/blob/4eca92cd7b87c2f1a1d6749266d8015f5d730448/lib/jsonnet/vm.rb#L9-L13 but that could be because I'm not too hot on the ruby c interface.

**Comment by kevindew** [comment]
I've switched `Jsonnet.parse` to `Jsonnet.evaluate` and rebased against master.

File: lib/jsonnet/evaluator.rb (hunk 1 above)
Comment (yugui) [inline lib/jsonnet/evaluator.rb]: Good point.

It's just because I was too lazy to call `#initialize` in C. It'd be clearer if the `vm_s_new` would call `#initialize`.
Let me try again.
Then, I'll merge this PR.

**Comment by yugui** [comment]
Merged.

Thank you for your contributions.
